README.md

```diff
@@ -55,6 +55,109 @@ fn main() {
 }
 ```
 
+# Extensions and Profiles
+
+EAR supports extension at top level (i.e. within the `Ear` struct), and also within
+`Appraisal`s. An extension is an additional field definition. Extensions can be defined by
+registering them with the `extensions` field of the corresponding struct. When registering an
+extension, you must provide a string name (used in JSON), an integer key (used in CBOR), and an
+`ExtensionKind` indicating which `ExtensionValue`s are valid.
+
+## Registering individual extensions
+
+Extensions can be registered directly with the corresponding struct's `extensions` field. Once
+they have been registered, their values can be set and queried
+
+```rust
+use ear::{Ear, Appraisal, ExtensionKind, ExtensionValue};
+
+let mut ear = Ear::new();
+ear.extensions.register("ext1", -1, ExtensionKind::String).unwrap();
+
+let mut appraisal = Appraisal::new();
+// extensions for Ear's and Appraisal's have their own namespaces, so it is
+// to use the same name/key in both.
+appraisal.extensions.register("ext1", -1, ExtensionKind::Integer).unwrap();
+
+ear.extensions.set_by_name(
+    "ext1",
+    ExtensionValue::String("foo".to_string()),
+).unwrap();
+
+appraisal.extensions.set_by_key(
+    -1,
+    ExtensionValue::Integer(42),
+).unwrap();
+
+ear.submods.insert("test".to_string(), appraisal);
+
+assert_eq!(
+   ear.extensions.get_by_key(&-1).unwrap(),
+   ExtensionValue::String("foo".to_string()),
+);
+
+assert_eq!(
+   ear.submods["test"].extensions.get_by_name("ext1").unwrap(),
+   ExtensionValue::Integer(42),
+);
+
+```
+
+Note: if you've obtained the `Ear` by deserializing from CBOR/JSON, `Extensions` struct
+will cache any values for any unexpected fields, so that when you register extensions
+afterwards, the corresponding unmarshaled values will be accessible.
+
+## Using Profiles
+
+Sets of extenions can be associted together within `Profile`s. A `Profile` can be registed,
+and can then be retrieved by its `id` when creating a new `Ear` or `Appraisal`
+
+```rust
+use ear::{Ear, Appraisal, ExtensionKind, ExtensionValue, Profile, register_profile};
+
+fn init_profile() {
+    let mut profile = Profile::new("my_profile");
+
+    profile.register_ear_extension("ext1", -1, ExtensionKind::String).unwrap();
+    profile.register_appraisal_extension("ext1", -1, ExtensionKind::Integer).unwrap();
```

Comment:
IIUC the registered extensions can appear in _any_ of the appraisal submods, right?  If so, this should be made explicit to avoid surprises.